disnake/ui/select/channel.py

```diff
@@ -0,0 +1,215 @@
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable, List, Optional, Tuple, Type, TypeVar, overload
+
+from ...components import ChannelSelectMenu
+from ...enums import ChannelType, ComponentType
+from ...utils import MISSING
+from .base import BaseSelect, P, V_co, _create_decorator
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from ...interactions.base import InteractionChannel
+    from ..item import DecoratedItem, ItemCallbackType, Object
+
+
+__all__ = (
+    "ChannelSelect",
+    "channel_select",
+)
+
+
+class ChannelSelect(BaseSelect[ChannelSelectMenu, "InteractionChannel", V_co]):
+    """Represents a UI channel select menu.
+
+    This is usually represented as a drop down menu.
+
+    In order to get the selected items that the user has chosen, use :attr:`.values`.
```

Comment:
not part of this pr, but the way to fix this state between different callbacks is a contextvar.